internal/resource/url.go

```diff
@@ -396,31 +395,43 @@ func (f *Fetcher) fetchFromS3(u url.URL, dest s3target, opts FetchOptions) error
 		defer cancelFn()
 	}
 
-	if f.AWSSession == nil {
+	if f.AWSConfig == nil {
 		var err error
-		f.AWSSession, err = session.NewSession(&aws.Config{
-			Credentials: credentials.AnonymousCredentials,
+		cfg, err := config.LoadDefaultConfig(ctx, func(o *config.LoadOptions) error {
+			o.Credentials = aws.AnonymousCredentials{}
+			return nil
 		})
 		if err != nil {
 			return err
 		}
+		f.AWSConfig = &cfg
 	}
-	sess := f.AWSSession.Copy()
+	cfg := f.AWSConfig.Copy()
 
 	// Determine the partition and region this bucket is in
 	regionHint := "us-east-1"
 	if f.S3RegionHint != "" {
 		regionHint = f.S3RegionHint
 	}
-	region, err := s3manager.GetBucketRegion(ctx, sess, u.Host, regionHint)
+	region, err := manager.GetBucketRegion(ctx, s3.NewFromConfig(cfg), u.Host)
 	if err != nil {
-		if aerr, ok := err.(awserr.Error); ok && aerr.Code() == "NotFound" {
-			return fmt.Errorf("couldn't determine the region for bucket %q: %v", u.Host, err)
+		var bnf manager.BucketNotFound
+		if errors.As(err, &bnf) {
+			f.Logger.Warning("couldn't determine the region for bucket %q: %v, trying again with region hint", u.Host, err)
+		}
+		// Use the region hint to determine the partition and region this bucket is in
+		region, err = manager.GetBucketRegion(ctx, s3.NewFromConfig(cfg), u.Host, func(o *s3.Options) {
+			o.Region = regionHint
+		})
+		if err != nil {
+			if errors.As(err, &bnf) {
+				return fmt.Errorf("couldn't determine the region for bucket %q with region hint %s: %v", u.Host, regionHint, err)
+			}
 		}
 		return err
 	}
 
-	sess.Config.Region = aws.String(region)
+	cfg.Region = *aws.String(region)
```

Comment:
Region is just a normal string here

Reply:
done, fixed in aws.go around regionHint too